src/components/richText/imageNode/ImageNode.tsx

```diff
@@ -0,0 +1,200 @@
+import {
+  $applyNodeReplacement,
+  DOMConversionMap,
+  DOMConversionOutput,
+  DOMExportOutput,
+  EditorConfig,
+  DecoratorNode,
+  LexicalNode,
+  NodeKey,
+  SerializedLexicalNode,
+  Spread,
+} from "lexical";
+
+import * as React from "react";
+
+import LexicalImage from "./LexicalImage";
+
+/**
+ * The raw Object obtained when first parsing a serialized {@link ImageNode}.
+ */
+export type SerializedImageNode = Spread<
+  {
+    altText: string;
+    height?: number;
+    maxWidth: number;
+    src: string;
+    width?: number;
+    type: "image";
+    version: 1;
+  },
+  SerializedLexicalNode
+>;
+
+/**
+ * Defines a Lexical Dev {@link DecoratorNode} for images.
+ */
+export class ImageNode extends DecoratorNode<JSX.Element> {
+  __src: string;
+  __altText: string;
+  __width: "inherit" | number;
+  __height: "inherit" | number;
+  __maxWidth: number;
+
+  constructor(
+    src: string,
+    altText: string,
+    maxWidth: number,
+    width?: "inherit" | number,
+    height?: "inherit" | number,
+    key?: NodeKey
+  ) {
+    super(key);
+    this.__src = src;
+    this.__altText = altText;
+    this.__maxWidth = maxWidth;
+    this.__width = width || "inherit";
+    this.__height = height || "inherit";
+  }
+
+  static getType(): string {
+    return "image";
+  }
+
+  static clone(node: ImageNode): ImageNode {
+    return new ImageNode(
+      node.__src,
+      node.__altText,
+      node.__maxWidth,
+      node.__width,
+      node.__height,
+      node.__key
+    );
+  }
+
+  getSrc(): string {
```

Comment:
What is getSrc? I assume getAltText and setWidthAndHeight just add an alt text label or affect styling.